src/main/java/com/google/sps/data/FirestoreDB.java

```diff
@@ -84,25 +84,56 @@ public long voteRecipe(String Id, int voteDiff) {
         getHidden ? DBUtils.tags() : DBUtils.tags().whereEqualTo(Tag.HIDDEN_KEY, false);
     return DBUtils.blockOnFuture(tagsQuery.get()).toObjects(Tag.class);
   }
+  
+  /** Returns a User object from userId. */
+    public User getUser(String userId) {
+      DocumentReference userRef = DBUtils.user(userId);
+      DocumentSnapshot user = DBUtils.blockOnFuture(userRef.get());
+      return user.toObject(User.class);
+    }
+   
+    /** Adds a User to the db based on an input userId. */
+    public String addUser(String userId) {
```

Comment:
Doesn't DBUtils.users().document() do the uid generation for us?
In which case, why are we trying to use an outside Id at all here?

Reply:
Yeah but we want to be able to use the decoded ID from the client token, not let firestore generate any old random id for the user

Reply:
Right forgot that auth generates uids as well